batch-models/src/main/scala/org/ekstep/analytics/updater/UpdateContentRating.scala

```diff
@@ -41,6 +43,20 @@ object UpdateContentRating extends IBatchModelTemplate[Empty, Empty, ContentMetr
   override def postProcess(data: RDD[ContentMetrics], config: Map[String, AnyRef])(implicit sc: SparkContext, fc: FrameworkContext): RDD[ContentMetrics] = {
     val baseURL = AppConf.getConfig("lp.system.update.base.url")
     if (data.count() > 0) {
+      val contentMetrics = data.filter(contentMetrics => contentMetrics.contentId != null && !contentMetrics.contentId.isEmpty)
+        .map(f => {
+          val contentMap = Map("me_averageRating" -> f.averageRating.getOrElse(0).toString,
+            "me_total_time_spent_in_app" -> f.totalTimeSpentInApp.getOrElse(0).toString,
+            "me_total_time_spent_in_portal" -> f.totalTimeSpentInPortal.getOrElse(0).toString,
+            "me_total_time_spent_in_desktop" -> f.totalTimeSpentInDeskTop.getOrElse(0).toString,
+            "me_total_play_sessions_in_app" -> f.totalPlaySessionCountInApp.getOrElse(0).toString,
+            "me_total_play_sessions_in_portal" -> f.totalPlaySessionCountInPortal.getOrElse(0).toString,
+            "me_total_play_sessions_in_desktop" -> f.totalPlaySessionCountInDeskTop.getOrElse(0).toString)
+          JSONUtils.serialize(GraphUpdateEvent("UPDATE_RATING",DateTime.now().getMillis,s"DP.${DateTime.now().getMillis}.${f.contentId}",Map("id"->"UPDATE_CONTENT_RATING","type"->"System"),Map("pdata"->Map("ver"->"1.0","id"->"org.ekstep.platform"),"channel"->s"channel-${f.contentId}"),Map("ver"->"1.0","id"->f.contentId),
+            Map("action"->"update-content-rating","stage"-> 4,"metadata"->contentMap))).replaceAll("objectInfo","object")
+        })
+      KafkaDispatcher.dispatch(config, contentMetrics)
```

Comment:
Don't use KafkaDispatcher directly in model. Use it as output dispatcher and take config from model-config similar to WFS.

Reply:
Done